vllm/worker/vineyard_llm_cache.py

```diff
@@ -80,23 +82,45 @@ def __init__(
                 f"max_num_batched_tokens ({self.max_num_batched_tokens}) must" \
                 f"be a multiple of chunk_size ({self.chunk_size})"
             )
-        self.buffer = torch.empty(
+        self.fetch_buffer, self.fetch_tensors = self._pinned_tensor_creator()
+        self.cuda_buffer = self.fetch_buffer.cuda()
+        # we use an object pool to reuse the pinned tensors and restrict the number of
+        # inflight tasks. if an update operation cannot get a tensor from the pool,
+        # meaning we already have VINEYARD_CACHE_MAX_INFLIGHT_TASKS tasks issued,
+        # it then simply skips the update. A completed task will return the used
+        # tensor back to the pool.
```

Comment:
This paragraph of comment probably should also go into the PR description, which should contain a brief summary of the motivation (i.e., issues to solve) and the high-level solution idea.

Reply:
added the paragraph into PR's desc